src/header/common/link.rs

```diff
@@ -0,0 +1,1027 @@
+            Ok(RelationType::Glossary)
+        } else if "help".eq_ignore_ascii_case(s) {
+            Ok(RelationType::Help)
+        } else if "hub".eq_ignore_ascii_case(s) {
+            Ok(RelationType::Hub)
+        } else if "index".eq_ignore_ascii_case(s) {
+            Ok(RelationType::Index)
+        } else if "last".eq_ignore_ascii_case(s) {
+            Ok(RelationType::Last)
+        } else if "latest-version".eq_ignore_ascii_case(s) {
+            Ok(RelationType::LatestVersion)
+        } else if "license".eq_ignore_ascii_case(s) {
+            Ok(RelationType::License)
+        } else if "next".eq_ignore_ascii_case(s) {
+            Ok(RelationType::Next)
+        } else if "next-archive".eq_ignore_ascii_case(s) {
+            Ok(RelationType::NextArchive)
+        } else if "payment".eq_ignore_ascii_case(s) {
+            Ok(RelationType::Payment)
+        } else if "prev".eq_ignore_ascii_case(s) {
+            Ok(RelationType::Prev)
+        } else if "predecessor-version".eq_ignore_ascii_case(s) {
+            Ok(RelationType::PredecessorVersion)
+        } else if "previous".eq_ignore_ascii_case(s) {
+            Ok(RelationType::Previous)
+        } else if "prev-archive".eq_ignore_ascii_case(s) {
+            Ok(RelationType::PrevArchive)
+        } else if "related".eq_ignore_ascii_case(s) {
+            Ok(RelationType::Related)
+        } else if "replies".eq_ignore_ascii_case(s) {
+            Ok(RelationType::Replies)
+        } else if "section".eq_ignore_ascii_case(s) {
+            Ok(RelationType::Section)
+        } else if "self".eq_ignore_ascii_case(s) {
+            Ok(RelationType::RelationTypeSelf)
+        } else if "service".eq_ignore_ascii_case(s) {
+            Ok(RelationType::Service)
+        } else if "start".eq_ignore_ascii_case(s) {
+            Ok(RelationType::Start)
+        } else if "stylesheet".eq_ignore_ascii_case(s) {
+            Ok(RelationType::Stylesheet)
+        } else if "subsection".eq_ignore_ascii_case(s) {
+            Ok(RelationType::Subsection)
+        } else if "successor-version".eq_ignore_ascii_case(s) {
+            Ok(RelationType::SuccessorVersion)
+        } else if "up".eq_ignore_ascii_case(s) {
+            Ok(RelationType::Up)
+        } else if "version-history".eq_ignore_ascii_case(s) {
+            Ok(RelationType::VersionHistory)
+        } else if "via".eq_ignore_ascii_case(s) {
+            Ok(RelationType::Via)
+        } else if "working-copy".eq_ignore_ascii_case(s) {
+            Ok(RelationType::WorkingCopy)
+        } else if "working-copy-of".eq_ignore_ascii_case(s) {
+            Ok(RelationType::WorkingCopyOf)
+        } else {
+            Ok(RelationType::ExtRelType(String::from(s)))
+        }
+    }
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// Utilities
+////////////////////////////////////////////////////////////////////////////////
+
+struct SplitAsciiUnquoted<'a> {
```

Comment:
Fantastic, good work separating the logic out to make it easier to follow exactly how parsing works.